src/main/java/com/powsybl/openloadflow/network/LfAction.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.openloadflow.graph.GraphConnectivity;
+import com.powsybl.openloadflow.network.impl.LfSwitch;
+import com.powsybl.security.action.Action;
+import com.powsybl.security.action.SwitchAction;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public class LfAction {
+
+    private final String id;
+
+    private LfBranch disabledBranch; // switch to open
+
+    private LfBranch enabledBranch; // switch to close
+
+    private LfNetwork network;
+
+    public LfAction(Action action, LfNetwork network) {
+        this.id = Objects.requireNonNull(action.getId());
+        this.network = Objects.requireNonNull(network);
+        switch (action.getType()) {
+            case "SWITCH":
+                SwitchAction switchAction = (SwitchAction) action;
+                LfBranch branch = network.getBranchById(switchAction.getSwitchId());
+                if (branch == null) {
+                    throw new PowsyblException("Branch " + switchAction.getSwitchId() + " not found in the network");
+                }
+                if (switchAction.isOpen()) {
+                    disabledBranch = branch;
+                } else {
+                    enabledBranch = branch;
+                }
+                break;
+            default:
+                throw new UnsupportedOperationException("Unsupported action type: "  + action.getType());
+        }
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public LfBranch getDisabledBranch() {
+        return disabledBranch;
+    }
+
+    public LfBranch getEnabledBranch() {
+        return enabledBranch;
+    }
+
+    public void apply() {
+        if (disabledBranch != null) {
+            disabledBranch.setDisabled(true);
+        }
+        if (enabledBranch != null) {
+            enabledBranch.setDisabled(false);
+            enabledBranch.getBus1().setDisabled(false);
+            enabledBranch.getBus2().setDisabled(false);
+        }
+
+        // update connectivity with disabled branches
+        List<LfBranch> disabledBranches = network.getBranches().stream()
+                .filter(LfElement::isDisabled)
+                .collect(Collectors.toList());
+        GraphConnectivity<LfBus, LfBranch> connectivity = network.getConnectivity();
+        connectivity.startTemporaryChanges();
+        disabledBranches.stream()
+                .filter(b -> b.getBus1() != null && b.getBus2() != null)
+                .filter(b -> !(b instanceof LfSwitch)) // FIXME
```

Comment:
@geofjamg I have to put this ugly and not working filter because I cannot cut twice the same branch... and all the switches to be potentially closed by a remedial action as already been cut after network loading.